ballerina/types.bal

```diff
@@ -69,12 +69,62 @@ public enum ColumnDelimiterEnum {
 
 # Represents the Salesforce client configuration.
 public type ConnectionConfig record {|
-    *config:ConnectionConfig;
     # The Salesforce endpoint URL
     string baseUrl;
     # Configurations related to client authentication
     http:BearerTokenConfig|config:OAuth2RefreshTokenGrantConfig|
         config:OAuth2PasswordGrantConfig|config:OAuth2ClientCredentialsGrantConfig auth;
+    # The HTTP version understood by the client
+    http:HttpVersion httpVersion = http:HTTP_2_0;
+    # Configurations related to HTTP/1.x protocol
+    ClientHttp1Settings http1Settings?;
```

Comment:
Any reason not to use `http:ClientHttp1Settings` [1] config here ? 

[1] - https://central.ballerina.io/ballerina/http/2.12.2#ClientHttp1Settings